packages/components/src/text-control/types.ts

```diff
@@ -14,27 +29,17 @@ export type TextControlProps = Pick<
 	/**
 	 * A function that receives the value of the input.
 	 */
-	onChange: ( value: string ) => void;
+	onChange: ( value: NoInfer< Value > ) => void;
```

Comment:
I was wondering about this, actually. I had assumed that maybe React does something special with the number input to make it work directly with numbers, but it seems like that assumption is wrong. Unfortunately, that means the PR does not make sense and should be closed.

Reply:
Ummm 🤔 Not sure if you have noticed it... In fact it's updated to make use of the `valueAsNumber` for `type="number"`:

<img width="1433" alt="image" src="https://github.com/user-attachments/assets/4d921190-c6aa-427b-ae6f-78a153d40b7e">

<img width="1264" alt="image" src="https://github.com/user-attachments/assets/0efacc49-5fe6-461e-bf35-b1945ecc2ae3">

And it should have great browser support: https://caniuse.com/mdn-api_htmlinputelement_valueasnumber

Reply:
Confirming that `valueAsNumber` browser support is good.

We should still make sure to support a `NaN` case, as it appears to be a potential return value (see [docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement#instance_properties)):

> `valueAsNumber`
> A number that represents the value of the element, interpreted as one of the following, in order: A time value, a number, or `NaN` if conversion is impossible.

A unit test to cover that scenario would be useful.

Reply:
Oh, sorry 🤦 But then this is a breaking change, and I don’t think we can ship it on such a widely used and established component, no? I can’t say this is unambiguously a “bug fix” rather than an enhancement.

Reply:
Yep, this is a big breaking change. I don't think there's nothing to do here unfortunately. At most expose the valueAsNumber attribute and make the generic depend on it. That seems quite convoluted though.